src/hooks/swap/useSwapInfo.tsx

```diff
@@ -64,10 +65,27 @@ function useComputeSwapInfo(routerUrl?: string): SwapInfo {
     useMemo(() => [currencyIn, currencyOut], [currencyIn, currencyOut])
   )
 
-  // Compute slippage and impact off of the trade so that it refreshes with the trade.
-  // (Using amountIn/amountOut would show (incorrect) intermediate values.)
   const slippage = useSlippage(trade.trade)
-  const { inputUSDC, outputUSDC, impact } = useUSDCPriceImpact(trade.trade?.inputAmount, trade.trade?.outputAmount)
+  const {
+    inputUSDC,
+    outputUSDC,
+    impact: stablecoinPriceImpact,
```

Comment:
I think this would be cleaner if you instead computed usdc as part of useSwapInfo, and that passed that into an updated `usePriceImpact` hook (to replace the existing `useUSDCPriceImpact` hook). The only reason the usdc prices are currently part of the price impact hook is to avoid computing them redundantly (because it is amongst the most expensive things we do in terms of compute) - adding actual logic to `useUSDCPriceImpact` makes me thing it'd be better to separate out the usdc computations.

eg

    const inputUSDC = useUSDCValue(inputAmount) ?? undefined
    const outputUSDC = useUSDCValue(outputAmount) ?? undefined
    const impact = usePriceImpact(trade.trade,  {inputUSDC, outputUSDC })